components/script/useChatSocket.tsx

```diff
@@ -190,11 +192,16 @@ const useChatSocket = (isEmpty?: boolean) => {
       }
 
       let confirmMessage = `Proceed with calling the ${frame.tool.name} tool?`;
-      if (frame.displayText) {
+      if (
+        !frame.tool.instructions?.startsWith('#!sys.openapi') &&
+        frame.displayText
+      ) {
         const tool = frame.tool?.name?.replace('sys.', '');
         confirmMessage = frame.tool?.source?.repo
           ? `Proceed with running the following (or allow all calls from the **${trimRepo(frame.tool?.source.repo!.Root)}** repo)?`
           : `Proceed with running the following (or allow all **${tool}** calls)?`;
+      } else if (frame.tool.instructions?.startsWith('#!sys.openapi')) {
```

Comment:
nit: if this if-block was first, then you wouldn't have to look at `frame.tool.instructions` in both places.